main/main.go

```
@@ -199,11 +212,85 @@ func usage() {
 	flag.PrintDefaults()
 }
 
+type diffToolStateType int
+
+const (
+	finStateInitial diffToolStateType = iota
+	dcpDriving      diffToolStateType = iota
+	finStateFinal   diffToolStateType = iota
+)
+
+type difftoolState struct {
+	state diffToolStateType
+	mtx   sync.Mutex
+}
+
+type xdcrDiffTool struct {
+	utils              xdcrUtils.UtilsIface
+	metadataSvc        service_def.MetadataSvc
+	remoteClusterSvc   service_def.RemoteClusterSvc
+	replicationSpecSvc service_def.ReplicationSpecSvc
+	logger             *xdcrLog.CommonLogger
+
+	specifiedRef  *metadata.RemoteClusterReference
+	specifiedSpec *metadata.ReplicationSpecification
+	filter        xdcrParts.FilterIface
+
+	sourceDcpDriver *dcp.DcpDriver
+	targetDcpDriver *dcp.DcpDriver
+
+	curState difftoolState
+}
+
+func NewDiffTool() *xdcrDiffTool {
+
+	difftool := &xdcrDiffTool{
+		utils: xdcrUtils.NewUtilities(),
+	}
+	difftool.metadataSvc, _ = metadata_svc.NewMetaKVMetadataSvc(nil, difftool.utils)
```

better not ignore error

---

done